apps/mobile/src/components/register-adoption-form/compose/form-steps/pet-upload-photos-step/pet-upload-photos-step.component.test.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import * as componentModule from './pet-upload-photos-step.component';
+import { usePetPhotosPicker } from './pet-upload-photos-step.hooks';
+import { fireEvent, renderWithProviders, waitFor } from '../../../../../test/test-utils';
+
+const { default: PetUploadPhotosStep } = componentModule;
+
+jest.mock('./pet-upload-photos-step.hooks', () => ({
+  usePetPhotosPicker: jest.fn(),
+}));
+
+const usePetPhotosPickerMock = (pickImage = jest.fn()) => {
+  (usePetPhotosPicker as jest.Mock).mockReturnValue({
+    images: [null, null, null],
+    pickImage,
+  });
+};
+
+describe('PetUploadPhotosStep', () => {
+  const pickImage = jest.fn();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders correctly', () => {
+    usePetPhotosPickerMock();
+
+    const { getByTestId } = renderWithProviders(<PetUploadPhotosStep />);
+
+    expect(getByTestId('Photo 1')).toBeTruthy();
+    expect(getByTestId('Photo 2')).toBeTruthy();
+    expect(getByTestId('Photo 3')).toBeTruthy();
+  });
+
+  it('calls pickImage when an image is pressed', () => {
+    usePetPhotosPickerMock(pickImage);
+
+    const { getByTestId } = renderWithProviders(<PetUploadPhotosStep />);
+
+    fireEvent.press(getByTestId('Photo 1'));
+    expect(pickImage).toHaveBeenCalledWith(0);
+    fireEvent.press(getByTestId('Photo 2'));
+    expect(pickImage).toHaveBeenCalledWith(1);
+    fireEvent.press(getByTestId('Photo 3'));
+    expect(pickImage).toHaveBeenCalledWith(2);
+  });
```

Comment:
How about:

```js
// place this at the top of the file
const IMAGE_SLOTS_IDENTIFIERS = ['first', 'second', 'third'];

describe.each(IMAGE_SLOTS_IDENTIFIERS)('when %s slot is pressed', (slot) => {
 const index = IMAGE_SLOTS_IDENTIFIERS.findIndex((item) => item === slot);

 it(`calls pickImage with ${index}`, () => {
  usePetPhotosPickerMock(pickImage);

  renderWithProviders(<PetUploadPhotosStep />);

  fireEvent.press(screen.getAllByHintText(`uploads the ${slot} pet picture`)[0]);
 });
});
```
By doing that, you can eliminate the `testId` and follow [testing library philosophy](https://testing-library.com/docs/dom-testing-library/api-accessibility/#testing-for-accessibility). See my comment below about accessibility properties.